misk/src/main/kotlin/misk/flags/Flags.kt

```diff
@@ -0,0 +1,207 @@
+package misk.flags
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import kotlin.properties.ReadOnlyProperty
+import kotlin.reflect.KClass
+import kotlin.reflect.KProperty
+
+interface Flag {
+    val name: String
+    val description: String
```

Comment:
(desc elsewhere vs. description here is my abbreviation annoyance. Don’t want to have to think about which to use)